src/Psy/ColorMode.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/*
+ * This file is part of Psy Shell.
+ *
+ * (c) 2012-2015 Justin Hileman
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Psy;
+
+/**
+ * A configuration object that represents one of 3 available color modes:
+ *   1. "auto" (the default): if color support is detected, output contains colors
+ *   2. "forced": force colors in output
+ *   3. "disabled": disable colors in output
+ */
+class ColorMode
```

Comment:
> It feels like all of these are twisting ourselves into contortions to work around PHP's lack of enums and the ability to typehint against them :)

Ha! Now that may very well be the case. :smile:

That said, I very much appreciate you entertaining my arguments, @bobthecow! Reworking the code now to use constants and equality.